src/graph/graphQLClient.ts

```diff
@@ -17,32 +17,62 @@ import config from '../config';
  * gives a graphQLClient for pocket-graph url
  *
  * This client initializes a `graphql-request` client
- * @param access_token accessToken of the user
- * @param consumer_key consumerKey assocuated with the user
+ * @param headers any headers received by proxy is just pass through to web graphQL proxy.
+ * @param accessToken accessToken of the user
+ * @param consumerKey consumerKey associated with the user
  */
-export function getClient(accessToken: string, consumerKey: string) {
-  return new GraphQLClient(
-    `${config.graphQLProxy}?consumer_key=${consumerKey}&access_token=${accessToken}`,
-    {
+export function getClient(
+  accessToken: string,
+  consumerKey: string,
+  headers: any
+) {
+  //these headers are not compatible with GraphQLClient's fetch.
+  //they throw an error instead, so ignoring them
+  // headers might include cookie, some observability traceIds, apollo studio headers
+  // so, we are implicitly passing them and removing only those that causes issues
+  delete headers['content-length'];
+  delete headers['content-type'];
+  delete headers['user-agent'];
+  delete headers['host'];
+  delete headers['accept-encoding'];
+  delete headers['connection'];
+
+  let url: string;
+
+  //to allow both access token/consumer key based auth or cookie based auth
+  if (accessToken && consumerKey) {
+    url = `${config.graphQLProxy}?consumer_key=${consumerKey}&access_token=${accessToken}`;
+  } else {
+    url = `${config.graphQLProxy}/`;
```

Comment:
this case throws an error 
https://pocket-dev.postman.co/workspace/Pocket~5b389395-697a-44ae-bcfd-3ff592fd249c/example/818037-6e6f5ce7-f71d-4675-b62e-7acd91e4000a - looks like a JSON parsing error, couldn't simulate this with local test as it gets back to fetch error if i mock a request